Packs/Legacy/Scripts/SetGridField/SetGridField.py

```diff
@@ -0,0 +1,200 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+
+def normalized_string(phrase: str) -> str:
+    """ Normalize columns or Grid to connected word in lower-case.
+
+    Args:
+        phrase: Phrase to normalize.
+
+    Returns:
+        str: Normalized phrase.
+
+    Examples:
+        >>> normalized_string("TestWord")
+        "testword"
+        >>> normalized_string("hello_world")
+        "hello_world"
+    """
+    return phrases_case.camel(phrase).replace("'", "").lower()
+
+
+def filter_the_dict(dict_obj: Dict[Any, Any], keys: List[str], max_keys: Optional[int] = None) -> Dict[Any, Any]:
+    """ Filter keys from Dictionary:
+            1. Will save all keys in keys parameter.
+            2. If key in index 0 is "*", will save all keys until max_keys (as much as Grid can include).
+
+    Args:
+        dict_obj: Dictionary to filter keys from.
+        keys: Keys to save.
+        max_keys: Max keys to save in case of keys[0] = "*"
+
+    Returns:
+        dict: Filtered dict.
+    """
+    new_dict = dict()
+    # Iterate over all the items in dictionary
+    if keys[0] != "*":
+        for (key, value) in dict_obj.items():
+            # Check if item satisfies the given condition then add to new dict
+            if key in keys:
+                new_dict[key] = value
+    else:
+        if max_keys:
+            new_dict = dict(list(dict_obj.items())[:max_keys])
+        else:
+            new_dict = dict_obj
+
+    return new_dict
+
+
+def get_current_table(grid_id: str) -> List[Dict[Any, Any]]:
+    """ Get current Data from the grid
+
+    Args:
+        grid_id: Grid ID to retrieve data from.
+
+    Returns:
+        list: Exsiting grid data.
+    """
+    current_table: Optional[List[dict]] = demisto.incidents()[0].get("CustomFields", {}).get(grid_id)
+    if current_table is None:
+        raise ValueError(f"The grid id isn't valid: {grid_id}")
+
+    return pd.DataFrame(current_table)
+
+
+def validate_entry_context(entry_context: Any, keys: List[str]):
+    """ Validate entry context structure is valid should be:
+            1. List[Dict[str, str]
+            2. Dict[str, str] - for developer it will be in first index of a list.
+
+    Args:
+        entry_context: Entry context to validate
+        keys: keys to collect data from
+
+    Raises:
+        ValueError: If structure is not valid.
+    """
+    exception_msg = "Not valid entry context path - dict[str,str] or list[dict[str,str]]"
+    if isinstance(entry_context, list):
+        for item in entry_context:
+            if not isinstance(item, dict):
+                raise ValueError(exception_msg)
+            else:
+                for key, value in item.items():
+                    if key in keys:
+                        if not (isinstance(key, str) and isinstance(value, str)):
+                            raise ValueError(exception_msg)
+    else:
+        raise ValueError(exception_msg)
```

Comment:
minimize indentations
```suggestion
    if not isinstance(entry_context, list):
        raise ValueError(exception_msg)

    for item in entry_context:
        if not isinstance(item, dict):
            raise ValueError(exception_msg)
            
        for key, value in item.items():
            if key in keys:
                if not (isinstance(key, str) and isinstance(value, str)):
                    raise ValueError(exception_msg)
```